lib/Telegram/Bot/Processor.pm

```diff
@@ -0,0 +1,183 @@
+                              my $loop = shift;
+                              my $now  = time();
+                              return unless ($now - $last_check) >= $seconds;
+                              $last_check = $now;
+                              $task->($self);
+                            });
+  };
+
+  push @{ $self->tasks }, $repeater;
+
+  $repeater->();
+}
+
+sub add_listener {
+  my $self    = shift;
+  my $coderef = shift;
+
+  push @{ $self->listeners }, $coderef;
+}
+
+sub init {
+  die "failed to init processor";
+}
+
+sub think {
+  my $self = shift;
+  $self->init();
+
+  $self->_add_getUpdates_handler;
+  Mojo::IOLoop->start unless Mojo::IOLoop->is_running;
+}
+
+sub getMe {
+  my $self = shift;
+  my $token = $self->token || croak "no token?";
+
+  my $url = "https://api.telegram.org/bot${token}/getMe";
+  my $api_response = $self->_post_request($url);
+
+  return Telegram::Bot::Object::User->create_from_hash($api_response, $self);
+}
+
+sub sendMessage {
+  my $self = shift;
+  my $args = shift || {};
+
+  my $send_args = {};
+  croak "no chat_id supplied" unless $args->{chat_id};
+  $send_args->{chat_id} = $args->{chat_id};
+
+  croak "no text supplied"    unless $args->{text};
+  $send_args->{text}    = $args->{text};
+
+  $send_args->{parse_mode} = $args->{parse_mode} if exists $args->{parse_mode};
+  $send_args->{disable_web_page_preview} = $args->{disable_web_page_preview} if exists $args->{disable_web_page_preview};
+  $send_args->{disable_notification} = $args->{disable_notification} if exists $args->{disable_notification};
+  $send_args->{reply_to_message_id}  = $args->{reply_to_message_id}  if exists $args->{reply_to_message_id};
+
+  my $token = $self->token || croak "no token?";
+  my $url = "https://api.telegram.org/bot${token}/sendMessage";
+  my $api_response = $self->_post_request($url, $send_args);
+
+  return Telegram::Bot::Object::Message->create_from_hash($api_response, $self);
+}
+
+sub forwardMessage {
+  my $self = shift;
+  my $args = shift || {};
+  my $send_args = {};
+  croak "no chat_id supplied" unless $args->{chat_id};
+  $send_args->{chat_id} = $args->{chat_id};
+
+  croak "no from_chat_id supplied"    unless $args->{from_chat_id};
+  $send_args->{from_chat_id}    = $args->{from_chat_id};
+
+  croak "no message_id supplied"    unless $args->{message_id};
+  $send_args->{message_id}    = $args->{message_id};
+
+  $send_args->{disable_notification} = $args->{disable_notification} if exists $args->{disable_notification};
+
+  my $token = $self->token || croak "no token?";
+  my $url = "https://api.telegram.org/bot${token}/forwardMessage";
+  my $api_response = $self->_post_request($url, $send_args);
+
+  return Telegram::Bot::Object::Message->create_from_hash($api_response, $self);
+}
+
+sub _add_getUpdates_handler {
+  my $self = shift;
+
+  my $http_active = 0;
+  my $last_update_id = -1;
+  my $token  = $self->token;
+
+  Mojo::IOLoop->recurring(0.1 => sub {
```

Comment:
please change 0.1 to 0.2

Reply:
0.2s timeout will affect user experience. Consider upgrading your computing power if you experience high cpu spikes

Reply:
We should migrate from perl to use modern data-fetching approaches. @dronov should choose new programming stack and rename repo accordingly. My preference would be brainfuck or rust.